Can't edit content after upgrading from version 21.04

I get this error repeatedly after upgrading from version 21.04 to 21.05: 

```
Caused by: com.dotmarketing.exception.DotDataException: ERROR: column "hash_ref" of relation "storage" does not exist
  Position: 45
	at com.dotmarketing.common.db.DotConnect.executeUpdate_aroundBody2(DotConnect.java:1227) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotmarketing.common.db.DotConnect$AjcClosure3.run(DotConnect.java:1) ~[dotcms_21.05_e1fa50b.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.9.2.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotmarketing.common.db.DotConnect.executeUpdate(DotConnect.java:1214) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotmarketing.common.db.DotConnect.executeUpdate(DotConnect.java:1198) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.storage.DataBaseStoragePersistenceAPIImpl.pushFileReference(DataBaseStoragePersistenceAPIImpl.java:502) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.storage.DataBaseStoragePersistenceAPIImpl.lambda$14(DataBaseStoragePersistenceAPIImpl.java:466) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.storage.DataBaseStoragePersistenceAPIImpl.wrapInTransaction(DataBaseStoragePersistenceAPIImpl.java:93) ~[dotcms_21.05_e1fa50b.jar:?]
	... 114 more
Caused by: org.postgresql.util.PSQLException: ERROR: column "hash_ref" of relation "storage" does not exist
  Position: 45
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2497) ~[postgresql-42.2.8.jar:42.2.8]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2233) ~[postgresql-42.2.8.jar:42.2.8]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:310) ~[postgresql-42.2.8.jar:42.2.8]
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:446) ~[postgresql-42.2.8.jar:42.2.8]
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:370) ~[postgresql-42.2.8.jar:42.2.8]
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:149) ~[postgresql-42.2.8.jar:42.2.8]
	at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:124) ~[postgresql-42.2.8.jar:42.2.8]
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-3.4.2.jar:?]
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-3.4.2.jar:?]
	at com.dotmarketing.common.db.DotConnect.executeUpdate_aroundBody2(DotConnect.java:1222) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotmarketing.common.db.DotConnect$AjcClosure3.run(DotConnect.java:1) ~[dotcms_21.05_e1fa50b.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.9.2.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotmarketing.common.db.DotConnect.executeUpdate(DotConnect.java:1214) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotmarketing.common.db.DotConnect.executeUpdate(DotConnect.java:1198) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.storage.DataBaseStoragePersistenceAPIImpl.pushFileReference(DataBaseStoragePersistenceAPIImpl.java:502) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.storage.DataBaseStoragePersistenceAPIImpl.lambda$14(DataBaseStoragePersistenceAPIImpl.java:466) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[dotcms_21.05_e1fa50b.jar:?]
	at com.dotcms.storage.DataBaseStoragePersistenceAPIImpl.wrapInTransaction(DataBaseStoragePersistenceAPIImpl.java:93) ~[dotcms_21.05_e1fa50b.jar:?]
	... 114 more
``` 

This error only happens when you start a fresh installation in version 21.04 and then try to upgrade to version 21.05 using postgres.

--- a/dotCMS/src/integration-test/java/com/dotcms/MainSuite.java
+++ b/dotCMS/src/integration-test/java/com/dotcms/MainSuite.java
@@ -94,6 +94,7 @@ import com.dotmarketing.startup.runonce.Task201102UpdateColumnSitelicTableTest;
 import com.dotmarketing.startup.runonce.Task210218MigrateUserProxyTableTest;
 import com.dotmarketing.startup.runonce.Task210319CreateStorageTableTest;
 import com.dotmarketing.startup.runonce.Task210321RemoveOldMetadataFilesTest;
+import com.dotmarketing.startup.runonce.Task210506UpdateStorageTableTest;
 import com.dotmarketing.util.ConfigTest;
 import com.dotmarketing.util.HashBuilderTest;
 import com.dotmarketing.util.TestConfig;
@@ -409,7 +410,8 @@ import org.junit.runners.Suite.SuiteClasses;
         ContentHandlerTest.class,
         ESIndexAPITest.class,
         FileAssetTemplateUtilTest.class,
-        SiteSearchJobImplTest.class
+        SiteSearchJobImplTest.class,
+        Task210506UpdateStorageTableTest.class
 })
 public class MainSuite {
 

--- a/dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task210506UpdateStorageTableTest.java
+++ b/dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task210506UpdateStorageTableTest.java
@@ -1,0 +1,36 @@
+package com.dotmarketing.startup.runonce;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.common.db.DotDatabaseMetaData;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import java.sql.SQLException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class Task210506UpdateStorageTableTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Test
+    public void testExecuteUpgrade() throws DotDataException, SQLException {
+        //Remove column if exists
+        final DotDatabaseMetaData dotDatabaseMetaData = new DotDatabaseMetaData();
+        if (dotDatabaseMetaData.hasColumn("storage", "hash_ref")) {
+            dotDatabaseMetaData
+                    .dropColumn(DbConnectionFactory.getConnection(), "storage", "hash_ref");
+        }
+
+        final Task210506UpdateStorageTable upgradeTask = new Task210506UpdateStorageTable();
+        assertTrue(upgradeTask.forceRun());
+        upgradeTask.executeUpgrade();
+        assertFalse(upgradeTask.forceRun());
+    }
+}

--- a/dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210506UpdateStorageTable.java
+++ b/dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210506UpdateStorageTable.java
@@ -1,0 +1,54 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotmarketing.common.db.DotDatabaseMetaData;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.util.Logger;
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * Upgrade task used to add column `hash_ref` in `storage` table if needed
+ */
+public class Task210506UpdateStorageTable extends AbstractJDBCStartupTask {
+
+    @Override
+    public boolean forceRun() {
+        try {
+            return !new DotDatabaseMetaData().hasColumn("storage", "hash_ref") ;
+        } catch (SQLException e) {
+            Logger.error(this, e.getMessage(),e);
+            return false;
+        }
+    }
+
+    @Override
+    public String getPostgresScript() {
+        return "ALTER TABLE storage ADD COLUMN hash_ref varchar(64);";
+    }
+
+    @Override
+    public String getMySQLScript() {
+        return "ALTER TABLE storage ADD hash_ref varchar(64);";
+    }
+
+    @Override
+    public String getOracleScript() {
+        return "ALTER TABLE storage ADD hash_ref varchar(64);";
+    }
+
+    @Override
+    public String getMSSQLScript() {
+        return "ALTER TABLE storage ADD hash_ref varchar(64);";
+    }
+
+    @Override
+    public String getH2Script() {
+        return null;
+    }
+
+    @Override
+    protected List<String> getTablesToDropConstraints() {
+        return null;
+    }
+}

--- a/dotCMS/src/main/java/com/dotmarketing/util/TaskLocatorUtil.java
+++ b/dotCMS/src/main/java/com/dotmarketing/util/TaskLocatorUtil.java
@@ -296,6 +296,7 @@ public class TaskLocatorUtil {
 		.add(Task210316UpdateLayoutIcons.class)
         .add(Task210319CreateStorageTable.class)
 		.add(Task210321RemoveOldMetadataFiles.class)
+        .add(Task210506UpdateStorageTable.class)
         .build();
         
         return ret.stream().sorted(classNameComparator).collect(Collectors.toList());

